docs/source/reference/northstar/safeio.rst

```diff
@@ -0,0 +1,89 @@
+
+    :param string file: The name of the file you want to store, this supports sub folders. Needs to be with the file type (e.g. ``/TitanData/tone.txt``).
+
+    :param string data: The saved data, this can be any valid String.
+
+Alternatively if you want a faster way to store :ref:`table_overview` you can use:
+
+.. cpp:function:: void NSSaveJSONFile(string file, table data)
+
+
+    :param string file: The name of the file you want to store, this supports sub folders. Doesn't have to be ``.json`` but will use the correct formatting for a ``.json``.
+
+    :param table data: The table that will be written to the file, this only supports the types specified in the :ref:`json_overview`.
+
+Loading a file
+--------------
+
+.. cpp:function:: void function NSLoadFile( string file, void functionref( string ) onSuccess, void functionref() onFailure = null )
+
+    :param string file: This is the name of the file you want to load, it has the same formating as in ``NSSaveFile``.
+
+    :param void functionref( string ) onSuccess: The function that gets exectued when the file is successfully loaded, the parameter ``string`` is the content of the loaded file.
+
+    :param void functionref() onFailure = null: The function that gets exectued when the loading was NOT successful, by default the function is just ``null``.
+
+    .. note::
+        If you are having trouble with functionrefs you can read up on them here: :ref:`functionref_overview`
+
+You can also get all saved file:
+
+.. cpp:function:: array<string> function NSGetAllFiles( string path = "" )
+
+    :param string path = "": Gets all files in a specified path, by default its just ``<profile>/saves/<mod directory name>``.
+
+    :returns: An array with all file names in the specified path.
+
+Deleting a file
+---------------
+
+.. cpp:function:: void NSDeleteFile(string file)
+
+    :param string file: This is the name of the file you want to check exsits, it has the same formating as in ``NSSaveFile``.
+
+
+File checks
+-----------
+
+.. cpp:function:: bool NSDoesFileExist(string file)
+
+    :param tring file: This is the name of the file you want to check exsits, it has the same formating as in ``NSSaveFile``.
+
+    :returns: ``true`` if the file was found, otherwise it returns ``false``.
+
+
+.. cpp:function:: int NSGetFileSize(string file)
+
+    :param string file: This is the name of the file you want to get the file size from.
+
+    :returns: Byte size of the specified file.
+
+    .. warning::
+        This fucntion will raise an error when the file doesnt exist.
+
+
+.. cpp:function:: bool NSIsFolder(string path)
+
+    :param string file: This is the path you want to check.
+
+    :returns: ``true`` if the path is a folder, otherwise it returns ``false``.
+
+.. cpp:function:: int NSGetTotalSpaceRemaining()
+
+    :returns: Amount of bytes you have left to write on.
+
+    .. note::
+        The max size of data you can store is ``50MB`` per mod.
```

Comment:
optionally, this can be set with a launch parameter `-maxfoldersize`, which actually uses _bytes_.